thorlcr/activities/external/thexternalslave.cpp

```diff
@@ -0,0 +1,175 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2012 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "jlib.hpp"
+#include "jset.hpp"
+#include "jqueue.tpp"
+#include "commonext.hpp"
+
+#include "thormisc.hpp"
+#include "thexception.hpp"
+#include "thbufdef.hpp"
+#include "thalloc.hpp"
+#include "eclrtl_imp.hpp"
+
+#include "thexternalslave.ipp"
+
+/////
+///////////////////
+//
+// ExternalSlaveActivity
+//
+
+class NodeActivityContext : public IThorActivityContext
```

Comment:
We should agree and try to be consistent, and I don't mind changing style, but all Thor classes begin with C, so e.g. CNodeActivityContext

Reply:
I will change.
I think I use CX for a class that is the only concrete implementation of an interface.  If there is more than one, then I often don't have a C prefix.